src/main/java/com/enablebanking/oidc/AuthController.java

```diff
@@ -37,35 +40,64 @@ public RedirectView auth(
             return new RedirectView(redirectUri + "?error=access_denied&error_description=" + URLEncoder.encode("scope \"openid\" is not present", UTF_8));
         }
 
-        return new RedirectView("/selectAccount" + "?nonce=" + nonce + "&redirect_uri=" + redirectUri + "&isPsd2=" + scopes.contains("psd2"));
+        if (state == null) {
+            // If state is not provided then generate a random one to avoid breaking the flow
+            state = authService.generateState();
```

Comment:
we should not do this. state is option parameter it should be returned only when passed to /authorize endpoint. as per open id connect spec

Reply:
@botsman could you please fix this?